app/controllers/laa_references_controller.rb

```diff
@@ -1,14 +1,27 @@
 # frozen_string_literal: true
 
 class LaaReferencesController < ApplicationController
+  before_action :set_defendant_asn_if_required,
+                :load_and_authorize_search
+  before_action :set_defendant_if_required
+
   def create
     authorize! :create, :link_maat_reference, message: I18n.t('unauthorized.default')
-    if link_laa_reference
-      flash[:notice] = I18n.t('laa_reference.link.success')
+    set_link_attempt
```

Comment:
use before_action for this

```
before_action :set_link_attempt
```

if you **need** to limit it to certain actions you can use `, only: :create`